### PR TITLE
added versions of pipelines to sharedConfig + working on startPipeline v2

### PR DESCRIPTION
**Pull request by RoanKanninga**

### CODE REVIEW

File: bin/startPipelineV2.sh [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+    exit 0
+}
+
+
+#
+# Source config files.
+#
+
+generateScripts () {
+	_sampleType=$1 ## DNA or RNA
```
Comment (pneerincx) [inline bin/startPipelineV2.sh]: * Local variables need to declare with local; without local in front of them they will be global instead.
* Please use ${var} (with curly brackets)
* Please quote variables.
E.g.:
     local _sampleType="${1}"

File: bin/startPipelineV2.sh [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+	_filePrefix=$2 ## name of the run sequencingStartDate_Sequencer_run_flowcell
+	_specie=$3 ##
```
Comment (pneerincx) [inline bin/startPipelineV2.sh]: One script and two scripts, but one species and two species:
_specie -> _species

File: bin/startPipelineV2.sh [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+		_pathToPipeline=$EBROOTNGS_DNA
+
+	elif [ "${_sampleType}" == "RNA" ]
+		_version="${NGS_RNA_VERSION}"
+		_pathToPipeline=$EBROOTNGS_RNA
+
+	fi
+
+	module load ${_loadPipeline}/"${_version}" || log4Bash 'FATAL' ${LINENO} "${FUNCNAME:-main}" ${?} "Failed to load ${_loadPipeline} module."
+	_generateShScript=${TMP_ROOT_DIR}/generatedscripts/${_filePrefix}/generate.sh"
```
Comment (pneerincx) [inline bin/startPipelineV2.sh]: I'm confused about the ${_pathToPipeline}/generate_template.sh script. What is the difference between that one and  the automated*_generate_template.sh scripts in this repo?

File: bin/startPipelineV2.sh [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+then
+	counting=$(ls ${TMP_ROOT_DIR}/Samplesheets/*.csv | wc -l)
+
+	log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "Checking $counting files "
+
+	for i in $(ls ${TMP_ROOT_DIR}/Samplesheets/*.csv) 
+	do
+
+		csvFile=$(basename $i)
+		filePrefix="${csvFile%.*}"
```
Comment (pneerincx) [inline bin/startPipelineV2.sh]: Is "filePrefix" the way it is used here the same as "project"?

File: bin/startPipelineV2.sh [hunk 5]
```diff
@@ -0,0 +1,409 @@
...
+	_filePrefix=$2 ## name of the run sequencingStartDate_Sequencer_run_flowcell
+	_specie=$3 ##
+	_workflowOrPanel=$4
+	_build=$5
+	_run=$6
+	_loadPipeline="NGS_${_sampleType}"
+
+
+	if [ "${_sampleType}" == "DNA" ]
+	then
```
Comment (pneerincx) [inline bin/startPipelineV2.sh]: Please put the "then" after "if" and "do" after "while" or "for" on the same line for readability.

File: bin/startPipelineV2.sh [hunk 6]
```diff
@@ -0,0 +1,409 @@
...
+                #grep 'processJob' submit.sh | tr '"' ' ' | awk -v pro=$PROJECT '{OFS="\t"} {print $2,pro"_"$2,pro,"","",""}' >>  ${LOGDIR}/${_project}/jobsPerProject.tsv
+
+                #CURLRESPONSE=$(curl -H "Content-Type: application/json" -X POST -d "{"username"="${USERNAME}", "password"="${PASSWORD}"}" https://${MOLGENISSERVER}/api/v1/login)
+                #TOKEN=${CURLRESPONSE:10:32}
+                #curl -H "x-molgenis-token:${TOKEN}" -X POST -F"file=@${LOGDIR}/${_project}/jobsPerProject.tsv" -FentityName='status_jobs' -Faction=add -Fnotify=false https://${MOLGENISSERVER}/plugin/importwizard/importFile
+                #echo "curl -H "x-molgenis-token:${TOKEN}" -X POST -F"file=@${LOGDIR}/${_project}/jobsPerProject.tsv" -FentityName='status_jobs' -Faction=add -Fnotify=false https://${MOLGENISSERVER}/plugin/importwizard/importFile"
+##
+#######################################
+
+                sleep 10
```
Comment (pneerincx) [inline bin/startPipelineV2.sh]: why sleep?

File: bin/startPipelineV2.sh [hunk 7]
```diff
@@ -0,0 +1,409 @@
...
+
+		touch "${LOGDIR}"/"${_project}"/"${_project}".pipeline.started
+		echo "${_project} started" >> ${_logger}
+		log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' \
+			"${_project} started"
+		printf "Pipeline: ${pipeline}\nStarttime:`date +%d/%m/%Y` `date +%H:%M`\nProject: $PROJECT\nStarted by: $WHOAMI\n \
+		Host: \${HOSTN}\n\nProgress can be followed via the command squeue -u $WHOAMI on $HOSTN.\nYou will receive an email when the pipeline is finished!\n\nCheers from the GCC :)" \
+		| mail -s "NGS_DNA pipeline is started for project $PROJECT on `date +%d/%m/%Y` `date +%H:%M`" ${ONTVANGER}
+
+		sleep 40
```
Comment (pneerincx) [inline bin/startPipelineV2.sh]: why sleep?

File: bin/startPipelineV2.sh [hunk 8]
```diff
@@ -0,0 +1,409 @@
...
+# Check commandline options.
+#
+if [[ -z "${group:-}" ]]; then
+	log4Bash 'FATAL' "${LINENO}" "${FUNCNAME:-main}" '1' 'Must specify a group with -g.'
+fi
+if [[ -n "${dryrun:-}" ]]; then
+	log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' 'Enabled dryrun option for rsync.'
+fi
+
+
```
Comment (pneerincx) [inline bin/startPipelineV2.sh]: I'm missing some bits here:
* sourcing of config files.
* thereShallOnlyBeOne function using flock to make sure only one instance is running (per group).
* check to make sure this script is executed using the right user account: ${group-ateambot}

File: bin/startPipelineV2.sh [hunk 9]
```diff
@@ -0,0 +1,409 @@
...
+##
+#### Copy raw data from prm to tmp
+#############################################
+
+## Here the data should be copied (what used to be CopyRawDataToCluster)
+
+
```
Comment (pneerincx) [inline bin/startPipelineV2.sh]: We can either do this here or using an additional protocol added to the NGS_DNA and NGS_RNA pipelines. Both options have some pro's and con's. If we want to keep NGS_DNA and NGS_RNA relatively universal and not dependent on our storage layout with prm, tmp and scr folders, we should do data staging here...

File: bin/startPipelineV2.sh [hunk 10]
```diff
@@ -0,0 +1,409 @@
...
+                        then
+				build="b38"
+			fi
+
+			if [[ "${projectName}" == *"Lexogen"* ]]
+                        then
+				workflow="lexogen"
+                        fi
+
+                        workflow="hisat"
```
Comment (pneerincx) [inline bin/startPipelineV2.sh]: ? This overrides the "if" to check whether we need the lexogen workflow....

File: bin/startPipelineV2.sh (hunk 4 above)
Comment (RoanKanninga) [inline bin/startPipelineV2.sh]: yes that is correct

File: bin/startPipelineV2.sh (hunk 3 above)
Comment (RoanKanninga) [inline bin/startPipelineV2.sh]: the automated generate template was not an exact copy of the original generate template in the pipeline repo's.  You don't want to change on two (or with the RNA pipeline included, three ) sites everytime you want to edit something in the generate_template. The automated_*_generate_template.sh will be removed here

File: bin/startPipelineV2.sh (hunk 5 above)
Comment (RoanKanninga) [inline bin/startPipelineV2.sh]: We can argue on this..

File: bin/startPipelineV2.sh (hunk 6 above)
Comment (RoanKanninga) [inline bin/startPipelineV2.sh]: can be removed

File: bin/startPipelineV2.sh (hunk 7 above)
Comment (RoanKanninga) [inline bin/startPipelineV2.sh]: can be removed

File: bin/startPipelineV2.sh (hunk 10 above)
Comment (RoanKanninga) [inline bin/startPipelineV2.sh]: lol, oops

File: bin/startPipelineV2.sh (hunk 1 above)
Comment (RoanKanninga) [inline bin/startPipelineV2.sh]: check

**Pull request merged**